test/unit/compute_resources/ovirt_test.rb

```diff
@@ -24,4 +24,29 @@ class OvirtTest < ActiveSupport::TestCase
       assert_find_by_uuid_raises(ActiveRecord::RecordNotFound, cr)
     end
   end
+
+  describe "associating operating system" do
+    setup do
+      @operating_systems_xml = Nokogiri::XML(File.read('test/fixtures/ovirt_operating_systems.xml'))
+      @compute_resource = FactoryGirl.build(:ovirt_cr).tap do |compute_resource|
+        fog_client = mock
```

Comment:
I liked the original one as it covered more code that stubbing just the available_operating_systems. But I can see the benefits of the narrowed one as well. Updated.